File: Queue/QueueArray.py

class Queue:
    def __init__(self,size=10):
        self.size = size
        self.front = -1
        self.rear = -1
        self.Q = [0] * size
        
    def Display(self):
        for i in range(self.front+1, self.rear+1):
            print(self.Q[i], end=" ")
    
    def enqueue(self, val):
        if self.rear==self.size-1:
            print("Queue is full!")
        else:
            self.rear += 1
            self.Q[self.rear] = val
    
    def dequeue(self):
        x = -1
        if self.rear==self.front:
            print("Queue is empty!")
        else:
            self.front += 1
            x = self.Q[self.front]
        return x
    
if __name__=="__main__":
    q = Queue(5)
    q.enqueue(1)
    q.enqueue(2)
    q.enqueue(3)
    q.enqueue(4)
    q.enqueue(5)
    q.dequeue()
    q.dequeue()
    q.Display()

File: Queue/QueueArray.c
#include "stdio.h"
#include "stdlib.h"

struct Queue{
    int size;
    int front;
    int rear;
    int *Q;
};

void Create(struct Queue *q, int size){
    q->size = size;
    q->front =q->rear = -1;
    q->Q = (int *)malloc(q->size * sizeof(int));
}

void Display(struct Queue q){
    for (int i=q.front+1;i<=q.rear;i++){
        printf("%d ", q.Q[i]);
    }
}

void Enqueue(struct Queue *q, int val){
    if (q->rear==q->size-1)
        printf("Queue is full\n");
    else{
        q->rear++;
        q->Q[q->rear] = val;
    }
}

int Dequeue(struct Queue *q){
    int x = -1;
    if (q->rear==q->front)
        printf("Queue is empty!\n");
    else{
        q->front++;
        x = q->Q[q->front];
    }
    return x;
}

int main(){
    struct Queue q;
    Create(&q, 5);
    Enqueue(&q,1);
    Enqueue(&q,2);
    Enqueue(&q,3);
    Enqueue(&q,4);
    Enqueue(&q,5);
    Dequeue(&q);
    Dequeue(&q);
    Display(q);
    return 0;
}